### ISSUE TITLE
openresty/lua

### ISSUE THREAD

**Issue by rbtcollins** [opentracing-contrib/meta#33]
I'd like to suggest that openresty/lua be on the list of integrations you should actively seek contributors for. 

Whats the procedure for donating such an integration, were we to make one?

**Comment by yurishkuro** [comment]
I was not aware of OpenTracing API being available for Lua, that would be the first step before integrations.

**Comment by rbtcollins** [comment]
Both are indeed necessary steps :) Whats the procedure for donating such things though?

**Comment by yurishkuro** [comment]
the biggest challenge is to find some other people interesting/experts in the language, so that you can collaborate on the API design to make it idiomatic for the language. OTSC members can review from OT perspective, but I, for example, won't be able to offer much advice on the language features.

Aside from that, we can create a repo and you can start hacking at it.

**Comment by rbtcollins** [comment]
Just an updated, we've got the c++ based integration with nginx happening for us now, and are still looking into a native Lua implementation for more control; if/when we do that we'll come back here and request a repository once we have at least a POC working.

**Comment by james-callahan** [comment]
I'm working on an opentracing library for Lua now.

**Comment by frankgreco** [comment]
I would be extremely interested in this contribution!

**Comment by james-callahan** [comment]
> @james-callahan do you have a link to your library?

Currently working on it over here: https://github.com/Kong/opentracing-lua

**Comment by tarekrached** [comment]
is this issue moot in light of [opentracing-contrib/nginx-opentracing](https://github.com/opentracing-contrib/nginx-opentracing)?

**Comment by moonming** [comment]
@tarekrached no, the opentracing-OpenResty will be implemented in Lua, and nginx-opentracing is implemented in C++.
